draft-ietf-tls-esni.md

```diff
@@ -299,8 +295,42 @@ Client MAY cache the ESNIKeys for a particular domain based on the TTL of the
 Resource Record, but SHOULD NOT cache it based on the not_after value, to allow
 servers to rotate the keys often and improve forward secrecy.
 
-Note that the length of this structure MUST NOT exceed 2^16 - 1, as the
-RDLENGTH is only 16 bits {{RFC1035}}.
+Note that the length of this structure MUST NOT exceed 2^16 - 10, as the
+RDLENGTH is only 16 bits {{RFC1035}} and 9 octets are consumed by the
+ESNIKeys= prefix.
+
+## ESNIInclude
+
+The DNS {{RFC1035}} allows most names to act as an alias via the CNAME
+record type. However, the apex of a zone cannot be an alias. Domain
+owners who cooperate with other organizations for server operation
+sometimes directly publish the A or AAAA address information for the
+servers at the apex of their zone, while relying on the more flexible
+CNAME for other hosted names. Unfortunately, a similar approach does
+not work well for ESNIKeys because they are expected to be rotated
+frequently and thus become out of date if published directly in
+another zone.
```

Comment:
This is kind of unclear, but I think the point is that the server operator provides the domain operator with the A record which the domain operator then installs in the DNS, but this works badly for ESNI Keys because then the server operator cannot frequently rotate the ENSI Keys.